### PR TITLE
i18n(hi): add `sidebar.mdx`

### PR DESCRIPTION
**Pull request by at-the-vr**
#### Description

- add translation for [sidebar navigation](https://starlight.astro.build/hi/guides/sidebar/) in hindi language

### CODE REVIEW

**Comment by liruifengv** [comment]
Hello. @at-the-vr There have a new commit for sidebar.mdx: https://github.com/withastro/starlight/pull/1355

**Comment by at-the-vr** [comment]
thanks for the heads-up @liruifengv 🙌

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 1]
```diff
@@ -0,0 +1,541 @@
...
+
+### लिंक
+
+`label` और `link` गुणों वाले ऑब्जेक्ट का उपयोग करके किसी आंतरिक या बाहरी पृष्ठ पर एक लिंक जोड़ें।
+
+```js "label:" "link:"
+starlight({
+	sidebar: [
+		// CSS और स्टाइलिंग मार्गदर्शिका का लिंक।
+		{ label: 'CSS और स्टाइलिंग', link: '/guides/css-and-tailwind/' },
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:66]:
```suggestion
		{ label: 'CSS और स्टाइलिंग', link: '/hi/guides/css-and-tailwind/' },
```

We usually point to the proper version in this example so if the user clicks it by curiosity, it does not change its language. e.g. this [example](https://starlight.astro.build/pt-br/guides/sidebar/#links).

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+		{ label: 'Astro', link: 'https://astro.build/' },
+	],
+});
+```
+
+उपरोक्त कॉन्फ़िगरेशन निम्नलिखित साइडबार उत्पन्न करता है:
+
+<SidebarPreview
+	config={[
+		{ label: 'CSS और स्टाइलिंग', link: '/guides/css-and-tailwind/' },
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:77]:
```suggestion
		{ label: 'CSS और स्टाइलिंग', link: '/hi/guides/css-and-tailwind/' },
```

To match the previous suggestion.

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 3]
```diff
@@ -0,0 +1,541 @@
...
+			items: [
+				{ label: 'अवयवों', link: '/guides/components/' },
+				{ label: 'अंतर्राष्ट्रीयकरण (i18n)', link: '/guides/i18n/' },
+				// लिंक का एक नेस्टेड समूह।
+				{
+					label: 'स्टाइलिंग',
+					items: [
+						{ label: 'CSS', link: '/guides/css-and-tailwind/' },
+						{ label: 'Tailwind', link: '/guides/css-and-tailwind/' },
+						{ label: 'Shiki', link: '/guides/css-and-tailwind/' },
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:106]:
```suggestion
				{ label: 'अवयवों', link: '/hi/guides/components/' },
				{ label: 'अंतर्राष्ट्रीयकरण (i18n)', link: '/hi/guides/i18n/' },
				// लिंक का एक नेस्टेड समूह।
				{
					label: 'स्टाइलिंग',
					items: [
						{ label: 'CSS', link: '/hi/guides/css-and-tailwind/' },
						{ label: 'Tailwind', link: '/hi/guides/css-and-tailwind/' },
						{ label: 'Shiki', link: '/hi/guides/css-and-tailwind/' },
```

Prefixing the link for the same reason, e.g. this [example](https://starlight.astro.build/ko/guides/sidebar/#%EA%B7%B8%EB%A3%B9).

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 4]
```diff
@@ -0,0 +1,541 @@
...
+			label: 'मार्गदर्शिका',
+			items: [
+				{ label: 'अवयवों', link: '/guides/components/' },
+				{ label: 'अंतर्राष्ट्रीयकरण (i18n)', link: '/guides/i18n/' },
+				{
+					label: 'स्टाइलिंग',
+					items: [
+						{ label: 'CSS', link: '/guides/css-and-tailwind/' },
+						{ label: 'Tailwind', link: '/guides/css-and-tailwind/' },
+						{ label: 'Shiki', link: '/guides/css-and-tailwind/' },
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:129]:
```suggestion
				{ label: 'अवयवों', link: '/hi/guides/components/' },
				{ label: 'अंतर्राष्ट्रीयकरण (i18n)', link: '/hi/guides/i18n/' },
				{
					label: 'स्टाइलिंग',
					items: [
						{ label: 'CSS', link: '/hi/guides/css-and-tailwind/' },
						{ label: 'Tailwind', link: '/hi/guides/css-and-tailwind/' },
						{ label: 'Shiki', link: '/hi/guides/css-and-tailwind/' },
```

To match the previous suggestion.

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 5]
```diff
@@ -0,0 +1,541 @@
...
+
+लिंक, समूह और स्वत: उत्पन्न किए गए समूह अपने लेबल के आगे बैज प्रदर्शित करने के लिए `badge` गुण भी शामिल कर सकते हैं।
+
+```js {10,17}
+starlight({
+	sidebar: [
+		{
+			label: 'मार्गदर्शिका',
+			items: [
+				// "New" बैज वाला एक लिंक.
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:248]:
```suggestion
				// "नया" बैज वाला एक लिंक.
```

This was translated in the example above this one and translated in the visual output of this example so this should be translated here too.

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 6]
```diff
@@ -0,0 +1,541 @@
...
+				{
+					label: 'अवयवों',
+					link: '/guides/components/',
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:251]:
```suggestion
					link: '/hi/guides/components/',
```

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 7]
```diff
@@ -0,0 +1,541 @@
...
+					badge: 'New',
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:252]:
```suggestion
					badge: 'नया',
```

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 8]
```diff
@@ -0,0 +1,541 @@
...
+				},
+			],
+		},
+		// "Deprecated" बैज वाला एक स्वतः निर्मित समूह।
+		{
+			label: 'संदर्भ',
+			badge: 'Deprecated',
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:259]: Could we translate the 2 "deprecated" badge references too? Would it makes sense?

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 9]
```diff
@@ -0,0 +1,541 @@
...
+उपरोक्त कॉन्फ़िगरेशन निम्नलिखित साइडबार उत्पन्न करता है:
+
+<SidebarPreview
+	config={[
+		{
+			label: 'मार्गदर्शिका',
+			items: [
+				{
+					label: 'अवयवों',
+					link: '/guides/components/',
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:275]:
```suggestion
					link: '/hi/guides/components/',
```

To match the suggested change above.

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 10]
```diff
@@ -0,0 +1,541 @@
...
+starlight({
+	sidebar: [
+		{
+			label: 'मार्गदर्शिका',
+			items: [
+				// पीले "Experimental" बैज वाला एक लिंक।
+				{
+					label: 'अवयवों',
+					link: '/guides/components/',
+					badge: { text: 'Experimental', variant: 'caution' },
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:318]:
```suggestion
				// पीले "Experimental" बैज वाला एक लिंक।
				{
					label: 'अवयवों',
					link: '/guides/components/',
					badge: { text: 'Experimental', variant: 'caution' },
```

Would it make sense to translate this badge? What do you think?

**Review by HiDeoo — COMMENTED** [review]
Thanks for another great translation :raised_hands:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://hackmd.io/1KYnrpUQR0SAGTzHIUoUDg?view).

I left a few comments and suggestions, it's mostly around consistency regarding badge translations and updating some example links so that if the user clicks them by curiosity, we do not change the language on them (and these ones are not checked by the links validator). Except for that, it looks pretty good to me :+1:

**Comment by HiDeoo** [comment]
Thanks for the update :raised_hands: I just looked at all the changes you made and everything looks great.

I see that there are 6 remaining comments to address. Just wanted to mention them as I'm not sure if you missed them or if you are still working on them. If you also need help with those, let me know and I can help out.

**Comment by at-the-vr** [comment]
Thanks Hideoo for the major feedback 🙌 , `sidebar.mdx` was definitely a roller coaster

File: docs/src/content/docs/hi/guides/sidebar.mdx [hunk 11]
```diff
@@ -0,0 +1,541 @@
...
+					},
+				},
+			],
+		},
+	]}
+/>
+
+## अंतर्राष्ट्रीयकरण
+
+[BCP-47](https://www.w3.org/International/questions/qa-choosing-language-tags) भाषा टैग निर्दिष्ट करके प्रत्येक समर्थित भाषा के लिए लिंक या समूह लेबल का अनुवाद करने के लिए लिंक और समूह प्रविष्टियों पर `अनुवाद` संपत्ति का उपयोग करें, उदाहरण के लिए `"en"`, `"ar"`, या `"zh-CN"`, कुंजी के रूप में और अनुवादित लेबल मान के रूप में। `label` गुण का उपयोग डिफ़ॉल्ट लोकेल और बिना अनुवाद वाली भाषाओं के लिए किया जाएगा।
```
Comment (HiDeoo) [inline docs/src/content/docs/hi/guides/sidebar.mdx:389]:
```suggestion
[BCP-47](https://www.w3.org/International/questions/qa-choosing-language-tags) भाषा टैग निर्दिष्ट करके प्रत्येक समर्थित भाषा के लिए लिंक या समूह लेबल का अनुवाद करने के लिए लिंक और समूह प्रविष्टियों पर `translations` संपत्ति का उपयोग करें, उदाहरण के लिए `"en"`, `"ar"`, या `"zh-CN"`, कुंजी के रूप में और अनुवादित लेबल मान के रूप में। `label` गुण का उपयोग डिफ़ॉल्ट लोकेल और बिना अनुवाद वाली भाषाओं के लिए किया जाएगा।
```

One final suggestion related to the last changes: this references the property name so it should not be translated.

After that, I think it's merging time 🚀

**Pull request merged**